server/src/main/java/org/candlepin/auth/Principal.java

```diff
@@ -69,6 +71,31 @@ public boolean canAccess(Object target, SubResource subResource, Access access)
         return false;
     }
 
+    public boolean canAccessAll(Collection targets, SubResource subResource, Access access) {
+
+        if (targets == null || targets.isEmpty()) {
+            log.debug(
+                    "{} principal checking for {} access to sub-resource: {}." +
+                    " Access to null or resource tried",
+                    this.getClass().getName(), access, subResource);
+            return canAccess(null, subResource, access);
+        }
```

Comment:
What is the use case for specifying no targets but requiring subs resource access?